cmd/mt-whisper-importer-reader/main.go

```diff
@@ -0,0 +1,328 @@
+		"Write logs to terminal",
+	)
+	httpEndpoint = flag.String(
+		"http-endpoint",
+		"http://127.0.0.1:8080/chunks",
+		"The http endpoint to send the data to",
+	)
+	chunkSpanStr = flag.String(
+		"chunkspans",
+		"10min",
+		"List of chunk spans separated by ':'. The 1st whisper archive gets the 1st span, 2nd the 2nd, etc",
+	)
+	namePrefix = flag.String(
+		"name-prefix",
+		"",
+		"Prefix to prepend before every metric name, should include the '.' if necessary",
+	)
+	threads = flag.Int(
+		"threads",
+		10,
+		"Number of workers threads to process .wsp files",
+	)
+	skipUnfinishedChunks = flag.Bool(
+		"skip-unfinished-chunks",
+		true,
+		"Chunks that have not completed their chunk span should be skipped",
+	)
+	orgId = flag.Int(
+		"orgid",
+		1,
+		"Organization ID the data belongs to ",
+	)
+	whisperDirectory = flag.String(
+		"whisper-directory",
+		"/opt/graphite/storage/whisper",
+		"The directory that contains the whisper file structure",
+	)
+	readArchivesStr = flag.String(
+		"read-archives",
+		"*",
+		"Comma separated list of positive integers or '*' for all archives",
+	)
+	chunkSpans   []uint32
+	readArchives map[int]struct{}
+	printLock    sync.Mutex
+)
+
+func main() {
+	flag.Parse()
+
+	for _, chunkSpanStrSplit := range strings.Split(*chunkSpanStr, ":") {
+		chunkSpans = append(chunkSpans, dur.MustParseUNsec("chunkspan", chunkSpanStrSplit))
+	}
+
+	if *readArchivesStr != "*" {
+		readArchives = make(map[int]struct{})
+		for _, archiveIdStr := range strings.Split(*readArchivesStr, ",") {
+			archiveId, err := strconv.Atoi(archiveIdStr)
+			if err != nil {
+				throwError(fmt.Sprintf("Invalid archive id %s: %s", archiveIdStr, err))
+			}
+			readArchives[archiveId] = struct{}{}
+		}
+	}
+
+	fileChan := make(chan string)
+
+	wg := &sync.WaitGroup{}
+	for i := 0; i < *threads; i++ {
+		go processFromChan(fileChan, wg)
+	}
+
+	getFileListIntoChan(fileChan, wg)
+	wg.Wait()
+}
+
+func throwError(msg string) {
+	msg = fmt.Sprintf("%s\n", msg)
+	if *exitOnError {
+		panic(msg)
+	} else {
+		printLock.Lock()
+		fmt.Fprintln(os.Stderr, msg)
+		printLock.Unlock()
+	}
+}
+
+func log(msg string) {
+	if *verbose {
+		printLock.Lock()
+		fmt.Println(msg)
+		printLock.Unlock()
+	}
+}
+
+func processFromChan(files chan string, wg *sync.WaitGroup) {
+	client := &http.Client{}
+
+	for file := range files {
+		fd, err := os.Open(file)
+		if err != nil {
+			throwError(fmt.Sprintf("ERROR: Failed to open whisper file '%s': %s\n", file, err))
+			continue
+		}
+		w, err := whisper.OpenWhisper(fd)
+		if err != nil {
+			throwError(fmt.Sprintf("ERROR: Failed to open whisper file '%s': %s\n", file, err))
+			continue
+		}
+
+		log(fmt.Sprintf("Processing file %s", file))
+		b, err := getMetric(w, file).Encode()
+		if err != nil {
+			throwError(fmt.Sprintf("%s", err))
+		}
+
+		req, err := http.NewRequest("POST", *httpEndpoint, b)
+		if err != nil {
+			throwError(fmt.Sprintf("Cannot send request to http endpoint %s: %s", *httpEndpoint, err))
```

Comment:
this is an error constructing the request, not sending it.  if this err is not nil, we should always panic or abort because it's a bug in our code that we must look into.